src/textual/binding.py

```diff
@@ -20,6 +21,22 @@
     from textual.dom import DOMNode
 
 BindingType: TypeAlias = "Binding | tuple[str, str] | tuple[str, str, str]"
+"""The possible types of a binding found in the `BINDINGS` class variable."""
+
+BindingIDString: TypeAlias = str
+"""The ID of a Binding defined somewhere in the application.
+
+Corresponds to the `id` parameter of the `Binding` class.
+"""
+
+KeyString: TypeAlias = str
+"""A string that represents a key binding.
+
+For example, "x", "ctrl+i", "ctrl+shift+a", "ctrl+j,space,x", etc.
+"""
+
+Keymap = Mapping[BindingIDString, KeyString]
```

Comment:
Does this not need a `TypeAlias` ?

Reply:
I think that's only needed if you're doing a forward reference in the definition by assigning to a string, like this:

```python
Keymap = "Mapping[BindingIDString, KeyString]"
```

Type-checkers have no way to know that's a type-alias unless you tell them.

> TypeAlias is particularly useful on older Python versions for annotating aliases that make use of forward references, as it can be hard for type checkers to distinguish these from normal variable assignments

https://docs.python.org/3/library/typing.html#typing.TypeAlias